adr/003-mediation-fees.md

```diff
@@ -0,0 +1,146 @@
+
+Mediating nodes demand fees for their mediation services. These fees should not only incentivize running a Raiden node, but also help balancing the network. This can be done by reducing the mediation fees for payments which leave channels in a more balanced state than before the payment (possibly even to a negative fee).
+
+## Decision Drivers
+
+* Balancing the network should be incentivized
+* Raiden nodes will act selfishly
+* Mediation fees should be flexible enough to accommodate arbitrary business models for parties operation Raiden nodes.
+
+## Considered Options
+
+1. A fixed fee + a proportional fee (flat + prop * amount)
+2. A fixed fee + a proportional fee + an imbalance fee based on a fixed imbalance measure
+3. A fixed fee + a proportional fee + submitting points of a function describing the imbalance fee
+4. Submitting points of a single function (dependent on the payment amount) which includes all fees
+
+## Decision Outcome
+
+Chosen option: **Option 3**, but with some important additions (see below).
+
+Option 1 would be sufficient for the short very term, but does not incentivize rebalancing. Option 2 does not work because no objective balance measure could be found that is valid across all business models / user interests. Option 4 would work but require sending a changed fee schedule after every transfer or mediation. Since this information is broadcast, that's a non-negligible amount of traffic.
+
+### Positive Consequences
+
+* Adaptable to different business models
+* Both in- and outgoing channels can be rebalanced
+* Fully exhausting a channel can be punished more strongly than just moving into the wrong direction
+* Fee schedules only have to be changed when the node's strategy changes or the on-chain capacity changes via `deposit` or `withdraw`
+
+### Negative Consequences
+
+* Comparatively complex
+
+## Detailed analysis
+
+One important detail is that fees are not only calculated for the channel over which the mediation payment is sent, but also for the channel over which the payment is received. Without this, a node could not incentivize receiving through a certain channel, which is just as important for rebalancing as sending through the right channel. So the total mediation fee for a node is
+
+`MF_total = MF_in_channel + MF_out_channel`
+
+where for each channel
+
+`MF_channel = flat_fee + proportional_fee * amount + IP(C_before) - IP(C_after)`
+
+where `IP` is the "Imbalance Penalty" function and `C_before/after` are the channel capacities before and after the payment. The channel capacity is the node's free capacity as reported to the PFS with the UpdatePFS messages.
+
+The IP function is a function that describes how much a node is willing to pay to move away from an undesirable channel capacity. If a node prefers to have a channel capacity of 5 while the total capacity of that channel is 10 (so that it could mediate up to 5 tokens in both directions) the IP function might look like
```

Comment:
We need to clarify if IP is relative or absolute values, and if it's token or token weis.

Reply:
I'll add
> The IP function is a function that describes how much a node is willing to pay The IP function is a function that describes how much a node is willing to pay **(as absolute amount or RDN-wei)** to move ...

RDN-wei is not something we've used before, but at least it's not ambiguous.

Reply:
I think we call it Rei, for RDN wei

Reply:
I just noticed after commenting that we're not talking about RDN and edited my question.